geonode/layers/views.py

```diff
@@ -360,7 +360,13 @@ def layer_detail(request, layername, template='layers/layer_detail.html'):
     if settings.SOCIAL_ORIGINS:
         context_dict["social_links"] = build_social_links(request, layer)
 
-    return render_to_response(template, RequestContext(request, context_dict))
+    context_proxy = settings.PROXY_URL
+    # Don't double proxy if we are utilizing the internal pki proxy
+    if '/pki' in layer.ows_url:
```

Comment:
We should probably not use a hardcoded path here, but instead check if the 'pki' app is installed first, then get [its 'reverse' path](https://docs.djangoproject.com/en/1.8/ref/urlresolvers/) for the test match.